docs/metrics.md

```diff
@@ -3,35 +3,143 @@
 
-In `log` mode, Metrics are logged and can be aggregated by metric agents.
+**Log Mode**
+
+In `log` mode, metrics are logged and can be aggregated by metric agents.
 Metrics are collected by default at the following locations in `log` mode:
 
 * Frontend metrics - `log_directory/ts_metrics.log`
-* Backend metrics - `log directory/model_metrics.log`
+* Backend metrics - `log_directory/model_metrics.log`
 
 The location of log files and metric files can be configured in the [log4j2.xml](https://github.com/pytorch/serve/blob/master/frontend/server/src/main/resources/log4j2.xml) file
 
-In `prometheus` mode, all metrics are made available in prometheus format via the [metrics API endpoint](https://github.com/pytorch/serve/blob/master/docs/metrics_api.md).
+**Prometheus Mode**
+
+In `prometheus` mode, all metrics are made available in prometheus format via the [metrics API endpoint](metrics_api.md).
+
+## Getting Started with TorchServe Metrics
+
+TorchServe defines metrics in a [yaml](https://github.com/pytorch/serve/blob/master/ts/configs/metrics.yaml) file, including both frontend metrics (i.e. `ts_metrics`) and backend metrics (i.e. `model_metrics`).
+When TorchServe is started, the metrics definition is loaded in the frontend and backend cache separately.
+The backend flushes the metrics cache once a load model or inference request is completed.
+
+Dynamic updates between the frontend and backend are _not_ currently being handled.
+
+The `metrics.yaml` is formatted with Prometheus metric type terminology:
+
+```yaml
+dimensions: # dimension aliases
+  - &model_name "ModelName"
+  - &level "Level"
+
+ts_metrics:  # frontend metrics
+  counter:  # metric type
+    - name: NameOfCounterMetric  # name of metric
+      unit: ms  # unit of metric
+      dimensions: [*model_name, *level]  # dimension names of metric (referenced from the above dimensions dict)
+  gauge:
+    - name: NameOfGaugeMetric
+      unit: ms
+      dimensions: [*model_name, *level]
+  histogram:
+    - name: NameOfHistogramMetric
+      unit: ms
+      dimensions: [*model_name, *level]
+
+model_metrics:  # backend metrics
+  counter:  # metric type
+    - name: InferenceTimeInMS  # name of metric
+      unit: ms  # unit of metric
+      dimensions: [*model_name, *level]  # dimension names of metric (referenced from the above dimensions dict)
+    - name: NumberOfMetrics
+      unit: count
+      dimensions: [*model_name]
+  gauge:
+    - name: GaugeModelMetricNameExample
+      unit: ms
+      dimensions: [*model_name, *level]
+  histogram:
+    - name: HistogramModelMetricNameExample
+      unit: ms
+      dimensions: [*model_name, *level]
+```
 
-## Frontend Metrics
+
+Note that **only** the metrics defined in the **metrics configuration file** can be emitted to logs or made available via the [metrics API endpoint](metrics_api.md). This is done to ensure that the metrics configuration file serves as a central inventory of all the metrics that Torchserve can emit.
```

Comment:
"can be emitted to logs" => "can be emitted to model_metrics.log", otherwise dumps to model_log.log